python/bigipconfigdriver.py

```diff
@@ -42,19 +43,95 @@
 root_logger.addHandler(console)
 
 
-class RootFilter(logging.Filter):
+class PartitionNameError(Exception):
+    """Exception type for F5 resource name."""
+
+    def __init__(self, msg):
+        """Create partition name exception object."""
+        Exception.__init__(self, msg)
+
+
+class IPV4FormatError(Exception):
+    """Exception type for improperly formatted IPv4 address."""
+
+    def __init__(self, msg):
+        """Create ipv4 format exception object."""
+        Exception.__init__(self, msg)
+
+
+class ResponseStatusFilter(logging.Filter):
     def filter(self, record):
         return not record.getMessage().startswith("RESPONSE::STATUS")
 
 
-root_logger.addFilter(RootFilter())
+class CertFilter(logging.Filter):
+    def filter(self, record):
+        return "CERTIFICATE" not in record.getMessage()
+
+
+class KeyFilter(logging.Filter):
+    def filter(self, record):
+        return "PRIVATE KEY" not in record.getMessage()
+
+
+root_logger.addFilter(ResponseStatusFilter())
+root_logger.addFilter(CertFilter())
+root_logger.addFilter(KeyFilter())
+
+
+def list_diff_exclusive(list1, list2):
+    """Return items found only in list1 or list2."""
+    return list(set(list1) ^ set(list2))
+
+
+def ipv4_to_mac(ip_str):
+    """Convert an IPV4 string to a fake MAC address."""
+    ip = ip_str.split('.')
+    if len(ip) != 4:
+        raise IPV4FormatError('Bad IPv4 address format specified for '
+                              'FDB record: {}'.format(ip_str))
+
+    return "0a:0a:%02x:%02x:%02x:%02x" % (
+        int(ip[0]), int(ip[1]), int(ip[2]), int(ip[3]))
+
+
+def extract_partition_and_name(f5_partition_name):
+    """Separate partition and name components for a Big-IP resource."""
+    parts = f5_partition_name.split('/')
+    count = len(parts)
+    if f5_partition_name[0] == '/' and count == 3:
+        # leading slash
+        partition = parts[1]
+        name = parts[2]
+    elif f5_partition_name[0] != '/' and count == 2:
+        # leading slash missing
+        partition = parts[0]
+        name = parts[1]
+    else:
+        raise PartitionNameError('Bad F5 resource name encountered: '
+                                 '{}'.format(f5_partition_name))
+    return partition, name
+
+
+def log_sequence(prefix, sequence_to_log):
+    """Helper function to log a sequence.
+
+    Dump a sequence to the logger, skip if it is empty
+
+    Args:
+        prefix: The prefix string to describe what's being logged
+        sequence_to_log: The sequence being logged
+    """
+    if sequence_to_log:
+        log.debug(prefix + ': %s', (', '.join(sequence_to_log)))
+
 
 DEFAULT_LOG_LEVEL = logging.INFO
 DEFAULT_VERIFY_INTERVAL = 30.0
 
 
-class K8sCloudBigIP(CloudBigIP):
-    """K8sCloudBigIP class.
+class K8sCloudServiceManager():
```

Comment:
Can we rename this to CF now?

Reply:
python/bigipconfigdriver.py in cf-bigip-ctlr is now identical to that in k8s-bigip-ctlr. There is no unique, orchestration-specific behavior here anymore. Now that this is common code, we should remove all cloud references/naming and determine how best to make this a common module.

Reply:
This would be a good time to remove all 'kubernetes' and 'k8s' references. I don't think there is value to any orchestration prefix either and would be fine with "" instead.

Reply:
Agree.